src/ducks/recurrence/queries.js

```diff
@@ -17,11 +17,15 @@ export const queryRecurrencesTransactions = recurrences =>
 
 export const bundleTransactionsQueryConn = ({ bundle }) => {
   return {
-    query: () =>
-      queryRecurrenceTransactions(bundle)
+    query: () => {
+      const initialQDef = queryRecurrenceTransactions(bundle)
+      const qDef = initialQDef
         .sortBy([{ date: 'desc' }])
+        .where({ ...initialQDef.selector, date: { $gt: null } })
```

Comment:
The attributes will be automatically indexed  in this order:  `date` should be first to actually use the index. Otherwise, couchdb will do a full-scan to run it.

Reply:
Ok thanks, I've modified the indexing.